modules/sonobiBidAdapter.js

```diff
@@ -388,6 +390,69 @@ export function _getPlatform(context = window) {
   }
   return 'desktop';
 }
+/**
+ * Check for local storage
+ * Generate a UUID for the user if one does not exist in local storage
+ * Store the UUID in local storage for future use
+ * @return {object} firstPartyData - Data object containing first party information
+ */
+function loadOrCreateFirstPartyData() {
+  var localStorageEnabled;
+
+  var FIRST_PARTY_KEY = '_iiq_fdata';
+  var tryParse = function (data) {
+    try {
+      return JSON.parse(data);
+    } catch (err) {
+      return null;
+    }
+  };
+  var readData = function (key) {
+    if (hasLocalStorage()) {
+      return window.localStorage.getItem(key);
+    }
+    return null;
+  };
+  var hasLocalStorage = function () {
+    if (typeof localStorageEnabled != 'undefined') { return localStorageEnabled; } else {
+      try {
+        localStorageEnabled = !!window.localStorage;
+        return localStorageEnabled;
+      } catch (e) {
+        localStorageEnabled = false;
+      }
+    }
+    return false;
+  };
+  var generateGUID = function () {
+    var d = new Date().getTime();
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
+      var r = (d + Math.random() * 16) % 16 | 0;
+      d = Math.floor(d / 16);
+      return (c === 'x' ? r : (r & 0x3) | 0x8).toString(16);
+    });
+  };
+  var storeData = function (key, value) {
+    try {
+      if (typeof key === 'string' && key.startsWith('_iiq_fdata')) {
```

Comment:
This check is unnecessary. storeData is only called once, and always with the FIRST_PARTY_KEY